wit-0.3.0-draft/terminal.wit

```diff
@@ -170,13 +165,40 @@ interface terminal-input {
 /// Support for terminal output.
 ///
 /// This defines APIs for controlling "termios" attributes affecting output,
-/// as well as control codes and "ANSI"-style escape sequences for terminal
-/// output streams.
+/// as well as control codes and "ANSI"-style escape sequences for controlling
+/// the cursor, changing the appearance of text, and other output controls.
+///
+/// This specification requires implementations to parse the output bytestreams
+/// and translate escape sequences and control codes to ensure consistent
+/// behavior. Implementations shall not permit arbitrary escape sequences to be
+/// communicated between applications and virtual terminals.
+///
+/// All terminal output streams are encoded in UTF-8. No 8-bit "extended ASCII"
+/// control code encodings are recognized. Any bytes which are not valid UTF-8
+/// evoke a trap, following the [UTF-8 decoder] algorithm in the
```

Comment:
is trapping the best option? it could be very annoying when you do something like `cat input-binary.dat | tee output-binary.dat` and accidentally write arbitrary binary to stdout.

Reply:
I think you're right; I've now changed it to do U+FFFD replacement.